tlsfuzzer/messages.py

```diff
@@ -413,8 +413,17 @@ def generate(self, state):
             self.version = state.client_version
         if self.random:
             state.client_random = self.random
-        if self.session_id is None:
+        if self.session_id is None and state.session_id:
             self.session_id = state.session_id
+        if self.session_id is None and self.extensions \
+                and ExtensionType.supported_versions in self.extensions:
+            # in TLS 1.3, the server should reply with CCS (middlebox compat
+            # mode) only when client sends a session_id
+            self.session_id = getRandomBytes(16)
```

Comment:
why this is 16-octet long while the spec says "legacy_session_id: ... a client not offering a pre-TLS 1.3 session MUST generate a new 32-byte value"?